src/manifold/src/impl.cpp

```diff
@@ -902,4 +903,55 @@ SparseIndices Manifold::Impl::VertexCollisionsZ(
   else
     return collider_.Collisions<false, false>(vertsIn);
 }
+/*
+ * Returns the minimum gap between two manifolds. Returns a float between
+ * 0 and searchLength.
+ */
+float Manifold::Impl::MinGap(const Manifold::Impl& other,
+                             float searchLength) const {
+  ZoneScoped;
+  Vec<Box> faceBox;
+  Vec<uint32_t> faceMorton;
+
+  this->GetFaceBoxMorton(faceBox, faceMorton);
+  this->SortFaceBoxMorton(faceBox, faceMorton);
+
+  transform(autoPolicy(faceBox.size()), faceBox.begin(), faceBox.end(),
+            faceBox.begin(), [searchLength](const Box& box) {
+              return Box(box.min - glm::vec3(searchLength),
+                         box.max + glm::vec3(searchLength));
+            });
+
+  Vec<Box> faceBoxOther;
+  Vec<uint32_t> faceMortonOther;
+
+  other.GetFaceBoxMorton(faceBoxOther, faceMortonOther);
+  other.SortFaceBoxMorton(faceBoxOther, faceMortonOther);
+
+  Collider collider{faceBox, faceMorton};
```

Comment:
Actually, maybe this is a lot simpler - Try following how the Boolean op works, which also starts with the collider. The collider should already exist - try just using it instead of creating a new one. That seems faster and less likely to be wrong.

Reply:
Oh right, I initially thought I needed to create a new collider because the boxes needed to be expanded/sorted. (now I can see that I could've updated the boxes of the collider as well). But I could just expand the other manifold's boxes, and no extra sorting is needed at all then.